>From b5f07ba60f76030b09f2f44db3ebe0487fb4ff25 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Thu, 30 Jun 2016 14:08:49 -0400
Subject: [PATCH 1/2] dh_dkms should fail cleanly with unknown package version
 or name

In some cases, the dkms file is mis-written, or the dh_dkms file
parsing is too brittle to find the package name or version.

in these cases, dh_dkms proceeds through, but spews errors like the
following:

Use of uninitialized value $package_version in concatenation (.) or string at /usr/bin/dh_dkms line 150.
Use of uninitialized value $package_version in concatenation (.) or string at /usr/bin/dh_dkms line 152.
Use of uninitialized value $package_version in concatenation (.) or string at /usr/bin/dh_dkms line 154.

This patch catches the error and reports it cleanly as an error.
---
 debian/scripts/dh_dkms | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/debian/scripts/dh_dkms b/debian/scripts/dh_dkms
index f2644ad..f5a1c4b 100755
--- a/debian/scripts/dh_dkms
+++ b/debian/scripts/dh_dkms
@@ -147,6 +147,12 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		doit("sed", "-i", "s/#MODULE_VERSION#/$package_version/g", $name);
 	}
 
+        error "could not determine package name"
+          unless defined($package_name);
+
+        error "could not determine package version"
+          unless defined($package_version);
+
 	autoscript($package, "prerm", "prerm-dkms",
 	    "s/#MODULE_NAME#/$package_name/;s/#MODULE_VERSION#/$package_version/");
 	autoscript($package, "postinst", "postinst-dkms",
-- 
2.8.1

